File: apps/python/lib/fizzbuzz.py
from lib.application.fizz_buzz_list_command import FizzBuzzListCommand
from lib.application.fizz_buzz_value_command import FizzBuzzValueCommand
from lib.domain.type.fizz_buzz_type import FizzBuzzType


class FizzBuzz:
    def __init__(self, type_: FizzBuzzType) -> None:
        self._type = type_

    @property
    def type(self) -> FizzBuzzType:
        return self._type

    def generate(self, number: int) -> str:
        return FizzBuzzValueCommand(self._type, number).execute().value

    def generate_list(self, count: int) -> list[str]:
        result = FizzBuzzListCommand(self._type).execute()
        return [result.get(i).value for i in range(count)]

    def print_fizzbuzz(self, count: int) -> None:
        for item in self.generate_list(count):
            print(item)

File: apps/ruby/lib/fizz_buzz/fizz_buzz.rb
# frozen_string_literal: true

require_relative 'domain/model/fizz_buzz_value'
require_relative 'domain/model/fizz_buzz_list'
require_relative 'domain/type/fizz_buzz_type'
require_relative 'domain/type/fizz_buzz_type_01'
require_relative 'domain/type/fizz_buzz_type_02'
require_relative 'domain/type/fizz_buzz_type_03'
require_relative 'domain/type/fizz_buzz_type_name'
require_relative 'application/fizz_buzz_command'
require_relative 'application/fizz_buzz_value_command'
require_relative 'application/fizz_buzz_list_command'

class FizzBuzz
  def self.generate(number)
    FizzBuzzType.create(FizzBuzzType::TYPE_01).generate(number).to_s
  end

  def self.generate_list
    (1..100).map { |number| generate(number) }
  end
end
